src/java.base/unix/native/libnet/Inet4AddressImpl.c

```diff
@@ -372,12 +372,15 @@ ping4(JNIEnv *env, jint fd, SOCKETADDRESS *sa, SOCKETADDRESS *netif,
         icmp = (struct icmp *)sendbuf;
         icmp->icmp_type = ICMP_ECHO;
         icmp->icmp_code = 0;
-        // let's tag the ECHO packet with our pid so we can identify it
-        icmp->icmp_id = htons(pid);
+        // same result as downcasting the little-endian pid, although we are not longer
```

Comment:
I don't think this is true. When downcasting the pid (which is at this stage in Nework/Big-Endian Order), the host order will be considered. Assuming that the downcast is to `uint16_t`, which seems is what icmp_id takes, the 16 least significant bits will be considered (if the host machine is little-endian of course). This is different from the 16 bit right shift. Here is some example output from a short C program I wrote (output is in hex to more easily demonstrate bitwise ops):

```
pid size: 4
int size: 4

getpid() testing:
pid   = be67f
htonl = 7fe60b00
cast  = b00
>> 16 = 7fe6
```

Reply:
On that note, is it ok to ignore any bits that are shifted out? For example, `7fe60b00 >> 16` results in  `00007fe6` while the trailing `0b00` is ignored.

Reply:
Unless I'm mistaken, this shifting will result in a different ident (icmp_id) value when run in big-endian systems (which appears unintentional).  While it is largely not all that significant any more, I still feel that it would be simpler and more straightforward to just retain the ident value, and introduce a new local variable for the full-pid. ( the comparison of the reply will continue to check against the payload pid value, rather than that of the ident ).

Alternatively (and better still), retain the pid value and do the narrowing cast and htoXX at the use-sites.

The comment can be simplified also.